### PR TITLE
Setup MVC-like code structure

### PR DESCRIPTION
**Pull request by Aerendel**
This covers #14 

Features currently setup:

- Model / View / Controller decoupling
- Necessary abstractions over napari Viewer and other shared plugin functionality
- Simple Router for navigation from one screen to another
- Debug utilities for easier development
- Full demo including 3 views (mpp, workflow selection and workflow steps) with working examples of UI setup, handling user events and navigation

### CODE REVIEW

File: napari_aicssegmentation/_tests/util/lazy_test.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    def test_lazy_property(self):
+        class TestObject:
+            @lazy_property
+            def property1(self):
+                return object()
+
+        test_obj = TestObject()
+        obj_id = id(test_obj.property1)
+        for i in range(10):
+            assert obj_id == id(test_obj.property1)
```
Comment (schoinh) [inline napari_aicssegmentation/_tests/util/lazy_test.py:16]: Can you tell me how this test works? Why is the for loop necessary?

File: napari_aicssegmentation/util/debug_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            end_time = time.perf_counter()
+            run_time = end_time - start_time
+
+            log.debug(f"END {name}. Finished in {run_time:.4f} secs")
+            return value
+        except Exception as ex:
+            log.error("=============================================")
+            log.error("\n\n" + traceback.format_exc())
+            log.error("=============================================")
+            raise ex
```
Comment (schoinh) [inline napari_aicssegmentation/util/debug_utils.py:31]: Sweet, this will be really helpful, thanks

**Review by schoinh — APPROVED** [review]
Looks good to me, I just have some minor comments and questions.

**Review by yrkim98 — APPROVED** [review]

File: napari_aicssegmentation/controller/mpp_controller.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from napari_aicssegmentation.core._interfaces import IApplication
+from napari_aicssegmentation.view.mpp_view import MppView
+from aicssegmentation.core.pre_processing_utils import image_smoothing_gaussian_3d
+from napari_aicssegmentation.core.controller import Controller
+from napari_aicssegmentation.util.debug_utils import debug_class
+from ._interfaces import IMppController
```
[the remark this answers is not in the document]
Comment (Aerendel) [reply napari_aicssegmentation/controller/mpp_controller.py:6]: What do you mean by well-organized? Alphabetically ordered?

File: napari_aicssegmentation/_tests/core/view_manager_test.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        view2: MagicMock = create_autospec(View)
+        view2.get_layout.return_value = QVBoxLayout()
+
+        # Act / Assert
+        self._view_manager.load_view(view1)
+        assert self._view_manager.current_view == view1
+        self._view_manager.load_view(view2)
+        assert self._view_manager.current_view == view2
+
+    def test_load_view_happy_path(self):
```
[the remark this answers is not in the document]
Comment (Aerendel) [reply napari_aicssegmentation/_tests/core/view_manager_test.py:27]: @schoinh happy path = happy code path. Meaning you're testing the straightforward code path through the method that leads to a successful result, versus testing edge / error cases. Usually the "happy path" covers most of the method's code, and the edge/error cases are a smaller fraction of the code, but just as important to test.

File: napari_aicssegmentation/_tests/util/lazy_test.py (hunk 1 above)
Comment (Aerendel) [reply napari_aicssegmentation/_tests/util/lazy_test.py:16]: I'm testing lazy initialization on a property, meaning that the first access to the property should give me a new object instance, but any subsequent access needs to give me the same instance that was created earlier (not a new one every time). So I'm accessing the property several times and making sure the object's ID is always the same. Loop is probably overkill - I could have just done 2 asserts in a row, but why do 2 when you can do 10 :)

File: napari_aicssegmentation/controller/mpp_controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (schoinh) [reply napari_aicssegmentation/controller/mpp_controller.py:6]: For example in napari_aicssegmentation/_tests/controller/workflow_steps_controller_test.py the imports are grouped by where they're from ([standard library, third party libraries, local](https://www.python.org/dev/peps/pep-0008/#imports)) and they're alphabetically ordered

File: napari_aicssegmentation/_tests/util/lazy_test.py (hunk 1 above)
Comment (schoinh) [reply napari_aicssegmentation/_tests/util/lazy_test.py:16]: Haha ok thanks

**Pull request merged**